### ISSUE TITLE
podman exec does not join the container network namespace

### ISSUE THREAD

**Issue by TristanCacqueray** [containers/podman#12749]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

Running podman exec to join a container does not attach the container network and keep the host netns.
This happens when the target container is started with `--userns keep-id` and `--network container:other-container`.

**Steps to reproduce the issue:**

1. Start the other-container: `podman run --name other-container --rm -it ubi8`

2. Start the target container: `podman run --userns keep-id --network container:other-container --name target --rm -it ubi8`

3. Join the target container: `podman exec target cat /proc/net/dev`

**Describe the results you received:**

The list of the host network devices.

**Describe the results you expected:**

The list of the container namespace devices (e.g. lo and tap0).

**Additional information you deem important (e.g. issue happens only occasionally):**

Removing the `--userns` or the `--network` argument fix the issue, it only happens when both are used at the same time for the target container.

I was not able to bisect the error (for example using podman 3.1.2 and conmon version 2.0.29 on the same host does not fix the issue), but this used to work as expected last December on fedora-35. Running the steps to reproduce the issue on a fedora-31 also does not reproduce the issue.

**Output of `podman version`:**

```
Version:      3.4.4
API Version:  3.4.4
Go Version:   go1.16.8
Built:        Wed Dec  8 21:45:07 2021
OS/Arch:      linux/amd64
```

**Output of `podman info --debug`:**

This happens on a fresh fedora-35.

**Have you tested with the latest version of Podman and have you checked the Podman Troubleshooting Guide? (https://github.com/containers/podman/blob/master/troubleshooting.md)**

No

**Comment by giuseppe** [comment]
moved back to Podman

**Comment by rhatdan** [comment]
@giuseppe want to take a stab at blocking this?

**Comment by giuseppe** [comment]
I am fine if we block `--userns` within a pod, but I am not sure about blocking the command reported in the issue.

The user's intentions are clear with the command `podman run --userns keep-id --network container:other-container ...` and the kernel allows that, and we artificially block it.

@TristanCacqueray what is your use case for using both `--userns keep-id` and `--network container:other-container`?

**Comment by TristanCacqueray** [comment]
@giuseppe my use case is to run a web browser with access to the host filesystem (using `--userns keep-id` to avoid issue with writing user owned files) and access to a network namespace (created by another vpn container). Here is the full setup documentation: https://github.com/podenv/podenv/blob/main/docs/howtos/vpn.md

If podman is going to block this combo (as it used to if I remember correctly), then I would have to always create the network namespace with the userns, even if it is not necessary.

**Comment by rhatdan** [comment]
You could have created the pod with --userns=keep-id

**Comment by Joerg-L** [comment]
Im comming from discussion https://github.com/containers/podman/discussions/12970#discussioncomment-2041003
From my perspective, havein the option to use user namespace on containers in a pod is a required and not really confusing scenario.

Let me describe the scenario, we are facing right now:
We want to run Atlassian Bitbucket to Podman container.
With help on the liked disucssion, we were able to get the Container for Bitbucket itself up and running.

How ever, when running Bitbucket in a container, the integrated Elasticsearch must be replaced with an external one.
So we have setupped a second container with Elastiscsearch.

Normally I woul put both together into a Pod to ensure that both are getting started/stopped and so on.
In addition, have both in the same pod would allow to access the elastic bod from the bitbucket pod with http://localhost:9200.

As they are not running in a pod, we need to do the communication by Host IP.

The issue for running both in a Pod is: the User/Group ID's are different.

Bitbucket: Container runs as root, but application is running as UID 2003
Elasticsearch: Container runs as root, but application is running as UID 1000

My Host User has Id 1024 and the mounted volumes should be accessable for backup/monitoring etc.

This scenario can be also be extended for a
- 3rd container to running the database
- 4th container for running a reverse proxy

Hope that clarify, why having the option in place is important.

**Comment by rhatdan** [comment]
@giuseppe @TristanCacqueray Where are we with this one.  Do we currently block --userns and --pod in the same container?

**Comment by rhatdan** [comment]
container-selinux (main) $ podman pod create --name dan
8d323c4f791279185cca5bfcbadd7a5d7661b216df7c681635d3b859adfbdb2f
container-selinux (main) $ podman run --pod dan --userns=keep-id alpine echo hi
Error: --userns and --pod cannot be set together

Looks like we do. so closing.

**Comment by TristanCacqueray** [comment]
It seems like the original issue got fixed by crun-1.4.1. Thanks @giuseppe !
